confgenerator/confgenerator.go

```diff
@@ -119,6 +119,7 @@ func (uc *UnifiedConfig) GenerateOtelConfig(ctx context.Context, outDir string)
 		LogLevel:          uc.getOTelLogLevel(),
 		ReceiverPipelines: receiverPipelines,
 		Pipelines:         pipelines,
+		Extensions:        map[string]interface{}{"googleclientauth": map[string]interface{}{}},
```

Comment:
talked offline: we will only set the extension when the experimental feature is used.